ui/src/Drawer.tsx

```diff
@@ -9,23 +10,67 @@ import SearchResults from "./SearchResults";
 import { ScholarReaderContext } from "./state";
 
 const PDF_VIEWER_DRAWER_OPEN_CLASS = "drawer-open";
-
+const BLACK_LISTED_CLASS_NAME = "MuiTooltip-tooltip";
 export class Drawer extends React.PureComponent {
+  static contextType = ScholarReaderContext;
+  context!: React.ContextType<typeof ScholarReaderContext>;
+
   componentWillUnmount() {
     const { pdfViewer } = this.context;
-    if (pdfViewer !== undefined && pdfViewer !== null) {
+    if (pdfViewer != null) {
       pdfViewer.container.classList.remove(PDF_VIEWER_DRAWER_OPEN_CLASS);
     }
   }
 
+  /**
+   * XXX(zkirby): This is an unfortunate byproduct of the clickaway listener
+   * listening to *all* clicks outside of the drawer. Any developer wishing to 
+   * add a button that opens the drawer will need to add the class name of that element
+   * to the black listed class names list. 
```

Comment:
This is great! Task 1: Update this comment to describe briefly what this code does (as the intent is not clear from the code alone)---specifically, it only closes the drawer if the click does not land on another annotation, because the drawer should be left open if a user has selected another annotation.

Reply:
I know I'm really leaning on you to update these code comments, and I'll keep pushing. The rule of thumb is that comments are necessary---and the times I'll always ask for them---is when the rationale behind the code isn't clear from the code itself. So I'll always ask for a comment if either of these are true:

1. The code required some **nuanced implementation**, and it would be a tragedy if someone saw this code and decided to replace it with something simpler (like here)
2. The code **isn't self-describing**: Someone won't be able to tell what the code is doing, or why it's doing it, after 30 seconds of looking at the code

Without these clarifications, people will either not want to touch the code, or try to delete it and replace it with something simpler, falling into pitfalls we have already solved. I'm hoping this code base will be both maintainable and robust for years to come and many collaborators, so let's get this right!

Reply:
Gotcha :)! No problem, I can update this comment to better reflect the functionality. Just to be clear, this code *does* close the drawer if the user clicks on another annotation as that is the implementation we discussed. Should that not be the case? If not there will need to be some changes that are made.